crates/ecs/src/systems/mod.rs

```diff
@@ -596,7 +595,18 @@ impl<'components, Component: Debug + Send + Sync + 'static + Sized> Clone
     for WriteSegment<'components, Component>
 {
     fn clone(&self) -> Self {
-        panic!("Cloning a segment for a mutable SystemParameter is not allowed.")
+        Self {
+            slices: self
+                .slices
+                .iter()
+                .map(|slice| {
+                    let ptr = (*slice).as_ptr();
+                    let len = slice.len();
+                    // SAFETY: Write segments will only be cloned when intra-parallelization is disabled.
+                    unsafe { slice::from_raw_parts_mut(mem::transmute(ptr), len) }
```

Comment:
Is there somewhere we could add some assertions (or `debug_assert`s if they impact performance) that checks that this is maintained?